design/ceph/mon-storage-class.md

```diff
@@ -0,0 +1,167 @@
+zone, by this we can leverage the spread of mons for providing high
+availability. This could be a useful scenario for a datacenter with own storage
+class in contrast to ones provided by cloud providers.
+
+Rook will enable the mon deployment when requested by the admin through the
+usual Rook CRDs.
+
+## Architecture
+
+To support this scenario, we can do something similar to the mon deployment for
+[Stretch clusters](https://github.com/ceph/ceph/blob/master/doc/rados/operations/stretch-mode.rst).
+The operator will have zonal awareness, associating mon deployments based on
+zones configuration provided.
+
+The type of failure domain used for stretch clusters is commonly "zone", but can
+be set to a different failure domain.
+
+### Domain Failure Configuration
+
+The topology of the K8s cluster is to be determined by the admin, outside the scope of Rook.
+Rook will simply detect the topology labels that have been added to the nodes.
+
+If the desired failure domain is a "zone", the `topology.kubernetes.io/zone` label should
+be added to the nodes. Any of the [topology labels](https://rook.io/docs/rook/latest/ceph-cluster-crd.html#osd-topology)
+supported by OSDs can be used.
+
+In the minimum configuration, two nodes in each data zone would be labeled, while one
+node in the arbiter zone is required.
+
+For example:
+
+```yaml
+topology.kubernetes.io/zone=a
+topology.kubernetes.io/zone=a
+topology.kubernetes.io/zone=b
+topology.kubernetes.io/zone=b
+topology.kubernetes.io/zone=arbiter
+```
+
+### Rook Cluster Design
+
+The core changes to rook are to associate the mons with the required zones.
+
+- Each zone will have atleast one mon assigned using labels for nodeaffinity.
+
+The new configuration will support the `zones` configuration where the
+failure domains available must be listed and the arbiter zone is identified
+optionally for mon configuration.
+
+```yaml
+  mon:
+    count: 3
+    allowMultiplePerNode: false
+    stretchCluster:
+      # The ceph failure domain will be extracted from the label, which by default is the zone. The nodes running OSDs must have
+      # this label in order for the OSDs to be configured in the correct topology. For topology labels, see
+      # https://rook.io/docs/rook/latest/CRDs/ceph-cluster-crd/#osd-topology.
+      failureDomainLabel: topology.kubernetes.io/zone
+      # The sub failure domain is the secondary level at which the data will be placed to maintain data durability and availability.
+      # The default is "host", which means that each OSD must be on a different node and you would need at least two nodes per zone.
+      # If the subFailureDomain is set to "osd", the OSDs would be allowed anywhere in the same zone including on the same node.
+      # If set to "rack" or some other intermediate failure domain, those labels would also need to be set on the nodes where
+      # the osds are started.
+      subFailureDomain: host
+      zones:
```

Comment:
Looks like the new `zones` is indented as `spec.mon.stretchClusters.zones`.  Is it supposed to be `spec.mon.zones` ?